lk2nd/device/leds.c

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2025 Yang Xiwen <forbidden405@outlook.com> */
+
+#include <debug.h>
+#include <libfdt.h>
+
+#include <lk2nd/hw/gpio.h>
+
+#include "device.h"
+
+static int lk2nd_leds_init(const void *dtb, int node)
+{
+	int i = 0, subnode, len;
+	struct gpiol_desc desc;
+	const char *state;
+
+	dprintf(SPEW, " Initializing LEDs\n");
+	dprintf(SPEW, " | label | gpio        | state |\n");
+
+	fdt_for_each_subnode(subnode, dtb, node) {
+		state = fdt_getprop(dtb, subnode, "default-state", &len);
+		if (len < 0 || state[len - 1] != '\0') {
+			dprintf(CRITICAL, "leds: Failed to get property \"default-state\". Broken dtb?\n");
+			continue;
+		}
+
+		if (!strcmp(state, "on"))
+			gpiol_get(dtb, subnode, NULL, &desc, GPIOL_FLAGS_OUT_ASSERTED);
+		else if (!strcmp(state, "off"))
+			gpiol_get(dtb, subnode, NULL, &desc, GPIOL_FLAGS_OUT_DEASSERTED);
+		else if (!strcmp(state, "keep"))
+			; /* do nothing */
+		else {
```

Comment:
I'd strongly prefer if/elseif/else constructions to either all have brackets, or none to have. Let's put brackets in all cases here.

Reply:
I still prefer to make 'keep' a valid no-op rather than an error to remain consistancy with linux dt-binding.